### PR TITLE
Add logic for subgroup branching

### PR DESCRIPTION
**Pull request by bifurcation**

### CODE REVIEW

File: interop/configs/branch.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    ],
+
+    "with_extensions": [
+      {"action": "createGroup", "actor": "alice"},
+      {"action": "externalJoin", "actor": "alice", "joiner": "bob", "members": []},
+      {"action": "externalJoin", "actor": "alice", "joiner": "charlie", "members": ["bob"]},
+      {"action": "externalJoin", "actor": "alice", "joiner": "diana", "members": ["bob", "charlie"]},
+      {"action": "externalJoin", "actor": "alice", "joiner": "eliza", "members": ["bob", "charlie", "diana"]},
+      {"action": "branch", "actor": "alice", "members": ["bob", "charlie", "diana", "eliza"],
+       "extensions": [{"extension_type": 3, "extension_data": "AAAA"}, {"extension_type": 5, "extension_data": "AA=="} ]}
```
Comment (mulmarta) [inline interop/configs/branch.json:19]: I wonder if there's a way to check that extensions have been injected? Maybe an RPC `GetGCEs` where the actor returns `repeated Extension extensions`? I'm worried that implementation 1 can't generate GCEs and creates a branch without them, which cant be noticed by other implementations

File: interop/configs/branch.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{
+  "scripts": {
```
Comment (mulmarta) [inline interop/configs/branch.json:2]: What about other combinations: with extensions and external tree etc. (there are 8 combinations)? Probably doesn't matter much, just for completeness.

**Review by mulmarta — APPROVED** [review]
Passes in the combination wickr-mlspp!

File: interop/configs/branch.json (hunk 1 above)
Comment (bifurcation) [reply interop/configs/branch.json:19]: https://github.com/mlswg/mls-implementations/issues/154

File: interop/configs/branch.json (hunk 2 above)
Comment (bifurcation) [reply interop/configs/branch.json:2]: https://github.com/mlswg/mls-implementations/issues/155

**Pull request merged**